### PR TITLE
ENH: Fix logging + model registration for Amulet runs

### PR DESCRIPTION
**Pull request by peterhessey**
Adds functionality to properly log IE-DL outputs and register trained models when running jobs through [amulet](https://amulet-docs.azurewebsites.net/).

### CODE REVIEW

**Review by kenza-bouzid — APPROVED** [review]
LGTM

File: InnerEye/ML/run_ml.py [hunk 1]
```diff
@@ -571,8 +571,12 @@ def register_model(self,
                 run_to_register_on = RUN_CONTEXT
                 logging.info(f"Registering the model on the current run {run_to_register_on.id}")
             logging.info(f"Uploading files in {final_model_folder} with prefix '{artifacts_path}'")
-            final_model_folder_relative = final_model_folder.relative_to(Path.cwd())
-            run_to_register_on.upload_folder(name=artifacts_path, path=str(final_model_folder_relative))
+
+            if not is_amulet_job():
```
Comment (ant0nsc) [inline InnerEye/ML/run_ml.py:575]: I find it easier to read if the condition is not inverted - you anyway cover both cases?

**Review by ant0nsc — APPROVED** [review]

File: InnerEye/ML/run_ml.py (hunk 1 above)
Comment (peterhessey) [reply InnerEye/ML/run_ml.py:575]: Agreed + updated!

**Comment by peterhessey** [comment]
> /azp run

why did this work for you but not for me? 😢

**Pull request merged**